test/controllers/paye/PayeAtsMainControllerSpec.scala

```diff
@@ -24,68 +24,217 @@ import org.mockito.Matchers.{any, eq => eqTo}
 
-  val fakeAuthenticatedRequest = buildPayeRequest("/annual-tax-summary/paye/treasury-spending")
+class PayeAtsMainControllerSpec extends PayeControllerSpecHelpers with ControllerBaseSpec with JsonUtil {
 
-  def sut = new PayeAtsMainController(mockPayeAtsService, FakePayeAuthAction, mcc, inject[PayeTaxsMainView])
+  implicit val fakeAuthenticatedRequest = buildPayeRequest("/annual-tax-summary/paye/treasury-spending")
 
-  "AtsMain controller" should {
+  val taxYearPlus1: Int = taxYear + 1
+  val taxYearList = (taxYear to taxYearPlus1).toList
 
-    "return OK response" in {
+  lazy val multipleYearsView = inject[PayeMultipleYearsView]
+  lazy val mainView = inject[PayeTaxsMainView]
 
-      when(
-        mockPayeAtsService
-          .getPayeATSData(eqTo(testNino), eqTo(taxYear))(any[HeaderCarrier], any[PayeAuthenticatedRequest[_]]))
-        .thenReturn(Right(mock[PayeAtsData]))
+  def getSingleYearData: PayeAtsData =
+    parseData[PayeAtsData](
+      loadAndReplace("/paye_ats.json", Map("$nino" -> testNino.nino))
+    )
 
-      val result = sut.show(fakeAuthenticatedRequest)
+  def getMultiYearData: List[PayeAtsData] =
+    parseData[List[PayeAtsData]](
+      loadAndReplace("/paye_ats_multiple_years.json", Map("$nino" -> testNino.nino))
+    )
 
-      status(result) shouldBe 200
+  private def parseData[A](str: String)(implicit reads: Reads[A]): A = Json.parse(str).as[A]
 
-      val document = Jsoup.parse(contentAsString(result))
+  def sut(multipleYearsEnabled: Boolean): PayeAtsMainController =
+    new PayeAtsMainController(mockPayeAtsService, FakePayeAuthAction, mcc, mainView, multipleYearsView) {
+      override val isMultipleYearsEnabled: Boolean = multipleYearsEnabled
+    }
 
-      document.title should include(
-        Messages("paye.ats.index.html.title") + Messages("generic.to_from", taxYear.toString, (taxYear + 1).toString))
+  "AtsMain controller" when {
 
-      document.getElementById("index-page-description").text() shouldBe (Messages("paye.ats.index.html.lede"))
+    "multiple years is enabled" should {
 
-      document.getElementById("tax-services-link").text shouldBe (Messages("paye.ats.index.html.tax_spend_link"))
+      "return OK and show multiple years page" when {
 
-      document.getElementsByTag("p").get(1).text shouldBe (Messages("English | Cymraeg"))
-      document.getElementsByTag("p").get(2).text shouldBe (Messages("paye.ats.index.html.lede"))
-      document.getElementsByTag("p").get(3).text shouldBe (Messages("paye.ats.index.html.tax_calc_description"))
-    }
+        "multiple years of data are returned" in {
+
+          when(
+            mockPayeAtsService
+              .getPayeATSMultipleYearData(eqTo(testNino), eqTo(taxYear), eqTo(taxYearPlus1))(
+                any[HeaderCarrier],
+                any[PayeAuthenticatedRequest[_]]))
+            .thenReturn(Future.successful(Right(getMultiYearData)))
+
+          val result = sut(true).show(fakeAuthenticatedRequest)
```

Comment:
can we extract the true/false here or maybe two functions to create the two different types of services (one with multiple years shown and the other with one)?